### PR TITLE
zcnsc commands

### PR DESCRIPTION
**Pull request by Dmdv**
This PR will add new commands for ethereum client wallet management

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/zwalletcli/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a branch at [0chain/system_test](https://github.com/0chain/system_test)

### CODE REVIEW

**Comment by cnlangzi** [comment]
gosdk 1.4.0 is upgraded on https://github.com/0chain/zwalletcli/pull/78. so close this PR

**Comment by Dmdv** [comment]
@cnlangzi This should be merged because those are my updates from zcnsc, nothing to do with other changes

**Comment by Dmdv** [comment]
To complete review I need to update documentation

File: cmd/bridge-client-init.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	bridgeClientInit.PersistentFlags().String("password", "", "password be used to unlock private key stored in local storage")
+	bridgeClientInit.PersistentFlags().String("ethereumaddress", "", "client Ethereum address")
+	bridgeClientInit.PersistentFlags().String("bridgeaddress", "", "bridge contract address")
+	bridgeClientInit.PersistentFlags().String("wzcnaddress", "", "WZCN token address")
+	bridgeClientInit.PersistentFlags().String("ethereumnodeurl", "", "Ethereum Node URL (Infura/Alchemy)")
+	bridgeClientInit.PersistentFlags().Int64("gaslimit", 300000, "appr. gas limit to execute Ethereum transaction")
+	bridgeClientInit.PersistentFlags().Int64("value", 0, "value sent along with Ethereum transaction")
+	bridgeClientInit.PersistentFlags().Float64("consensusthreshold", 0.75, "Consensus threshold required to reach consensus for burn tickets")
+
+	_ = bridgeClientInit.MarkFlagRequired("password")
```
Comment (cnlangzi) [inline cmd/bridge-client-init.go]:
```suggestion
        bridgeClientInit.MarkFlagRequired("password") //nolint:errcheck
        bridgeClientInit.MarkFlagRequired("ethereumaddress") //nolint:errcheck
        bridgeClientInit.MarkFlagRequired("bridgeaddress") //nolint:errcheck
	bridgeClientInit.MarkFlagRequired("wzcnaddress") //nolint:errcheck
	bridgeClientInit.MarkFlagRequired("ethereumnodeurl") //nolint:errcheck
	bridgeClientInit.MarkFlagRequired("gaslimit") //nolint:errcheck
	bridgeClientInit.MarkFlagRequired("consensusthreshold") //nolint:errcheck
```

File: cmd/bridge-list-authorizers.go [hunk 2]
```diff
@@ -10,11 +12,18 @@ var listAuthorizers = &cobra.Command{
 	Short: "list authorizers",
 	Long:  `list available authorizers`,
 	Args:  cobra.MinimumNArgs(0),
-	Run: func(cmd *cobra.Command, args []string) {
+	Run: func(_ *cobra.Command, _ []string) {
```
Comment (cnlangzi) [inline cmd/bridge-list-authorizers.go]:
```suggestion
Run: func(*cobra.Command, []string) {
```

File: cmd/bridge-owner-init.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+	bridgeOwnerInit.PersistentFlags().String("password", "", "password be used to unlock private key stored in local storage")
+	bridgeOwnerInit.PersistentFlags().String("ethereumaddress", "", "client Ethereum address")
+	bridgeOwnerInit.PersistentFlags().String("bridgeaddress", "", "bridge contract address")
+	bridgeOwnerInit.PersistentFlags().String("wzcnaddress", "", "WZCN token address")
+	bridgeOwnerInit.PersistentFlags().String("ethereumnodeurl", "", "Ethereum Node URL (Infura/Alchemy)")
+	bridgeOwnerInit.PersistentFlags().Int64("gaslimit", 0, "appr. gas limit to execute Ethereum transaction")
+	bridgeOwnerInit.PersistentFlags().Int64("value", 0, "value sent along with Ethereum transaction")
+
+	_ = bridgeOwnerInit.MarkFlagRequired("password")
```
Comment (cnlangzi) [inline cmd/bridge-owner-init.go]:
```suggestion
        bridgeOwnerInit.MarkFlagRequired("password") //nolint:errcheck
	bridgeOwnerInit.MarkFlagRequired("ethereumaddress") //nolint:errcheck
	bridgeOwnerInit.MarkFlagRequired("bridgeaddress") //nolint:errcheck
	bridgeOwnerInit.MarkFlagRequired("wzcnaddress") //nolint:errcheck
	bridgeOwnerInit.MarkFlagRequired("ethereumnodeurl") //nolint:errcheck
	bridgeOwnerInit.MarkFlagRequired("gaslimit") //nolint:errcheck
```

File: cmd/ethereum-account-register.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(registerAccount)
+
+	registerAccount.PersistentFlags().String("mnemonic", "", "mnemonic")
+	registerAccount.PersistentFlags().String("password", "", "password")
+
+	_ = registerAccount.MarkFlagRequired("mnemonic")
```
Comment (cnlangzi) [inline cmd/ethereum-account-register.go]:
```suggestion
	registerAccount.MarkFlagRequired("mnemonic") //nolint:errcheck
	registerAccount.MarkFlagRequired("password") //nolint:errcheck
```

File: cmd/ethereum-accounts-list.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/0chain/gosdk/zcnbridge"
+	"github.com/spf13/cobra"
+)
+
+var listEthAccounts = &cobra.Command{
+	Use:   "eth-list-accounts",
+	Short: "list Ethereum account registered in local key chain",
+	Long:  `list available ethereum accounts`,
+	Args:  cobra.MinimumNArgs(0),
+	Run: func(_ *cobra.Command, _ []string) {
```
Comment (cnlangzi) [inline cmd/ethereum-accounts-list.go]:
```suggestion
Run: func(*cobra.Command, []string) {
```

**Review by cnlangzi — DISMISSED** [review]

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/default_client_config  |
| gosdk  | feature/default_client_config  |
| zboxcli  | staging  |
| zwalletcli  | feature/update_74  |
| 0chain  | feature/zcnsc_eventsdb_storage  |
| blobber  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/zcnsc_zwallet_update_74  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | feature/update_74  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: cmd/bridge-client-init.go (hunk 1 above)
Comment (Dmdv) [reply cmd/bridge-client-init.go]: Fixed

File: cmd/bridge-list-authorizers.go (hunk 2 above)
Comment (Dmdv) [reply cmd/bridge-list-authorizers.go]: Fixed

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/zcnsc_zwallet_update_74  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | feature/update_74  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/zcnsc_zwallet_update_74  |
| gosdk  | feature/default_client_config  |
| zboxcli  | staging  |
| zwalletcli  | feature/update_74  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/zcnsc_zwallet_update_74  |
| gosdk  | feature/default_client_config  |
| zboxcli  | staging  |
| zwalletcli  | feature/update_74  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by stewartie4** [comment]
approving on behalf of @cnlangzi / @Sriep as per https://0chain.slack.com/archives/C02AV6MKT36/p1644104715711039?thread_ts=1644101141.278579&cid=C02AV6MKT36

**Review by stewartie4 — APPROVED** [review]

**Pull request merged**